mscore/newwizard.cpp

```diff
@@ -357,6 +327,53 @@ QString NewWizardPage4::templatePath() const
       return path;
       }
 
+//---------------------------------------------------------
+//   setTemplateLayout
+//---------------------------------------------------------
+
+void NewWizardPage4::setTemplateLayout()
+      {
+      if (layout()) {
+            QLayout* oldLayout = layout();
+            qDeleteAll(oldLayout->children());
+            delete oldLayout;
+            }
+
+      templateFileBrowser->setStripNumbers(true);
+      QDir dir(mscoreGlobalShare + "/templates");
+      QFileInfoList fil = dir.entryInfoList(QDir::NoDotAndDotDot | QDir::Readable | QDir::Dirs | QDir::Files, QDir::Name);
+      if(fil.isEmpty()){
+          fil.append(QFileInfo(QFile(":data/Empty_Score.mscz")));
+          }
+
+      QDir myTemplatesDir(preferences.getString(PREF_APP_PATHS_MYTEMPLATES));
+      fil.append(myTemplatesDir.entryInfoList(QDir::NoDotAndDotDot | QDir::Readable | QDir::Dirs | QDir::Files, QDir::Name));
+
+      templateFileBrowser->setShowCustomCategory(true);
+      templateFileBrowser->setScores(fil);
+      templateFileBrowser->setSizePolicy(QSizePolicy(QSizePolicy::Ignored, QSizePolicy::Ignored));
+
+      QVBoxLayout* newLayout = new QVBoxLayout;
+      QHBoxLayout* searchLayout = new QHBoxLayout;
+      QLineEdit* search = new QLineEdit;
+      search->setPlaceholderText(tr("Search"));
+      search->setClearButtonEnabled(true);
+      search->setSizePolicy(QSizePolicy::Preferred, QSizePolicy::Fixed);
+      searchLayout->addSpacerItem(new QSpacerItem(0, 0, QSizePolicy::MinimumExpanding, QSizePolicy::Maximum));
+      searchLayout->addWidget(search);
+
+      newLayout->addLayout(searchLayout);
+      newLayout->addWidget(templateFileBrowser);
+
+      setLayout(newLayout);
+
+      connect(templateFileBrowser, SIGNAL(scoreSelected(const QString&)), SLOT(templateChanged(const QString&)));
```

Comment:
Are you sure that connects are required on each `setTemplateLayout()` call? Previously, it was set once in ctor. Now, it is set each update.

Reply:
I'm not sure as I'm not too familiar with QT's GUI classes.  For one, the connects require variables that are updated within the update function (search).  Also, I read [here](http://doc.qt.io/qt-5/qobject.html#dtor.QObject) that for a QObject's deconstructor:

> All signals to and from the object are automatically disconnected, and any pending posted events for the object are removed from the event queue. 

which is why I reconnected every time, although I may have assumed wrong.

Reply:
I've tested not connecting on each update and the layouts don't update correctly when connecting only once in the constructor @anatoly-os